jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannelState.java

```diff
@@ -559,19 +576,25 @@ public void errorHandling()
         }
     }
 
-    public void errorHandlingComplete()
+    public void errorHandlingComplete(Throwable failure)
```

Comment:
some javadoc explaining when this is called would be worthwhile.  Specifically it is called when the callback passed to the ErrorHandler.handle method has been completed.    This is either because a core handler has completed handling the response;
OR the request has been passed to a servlet and it has completed the response;
OR the request has been passed to a servlet and it changed state with `startAsync` or `sendError`, so we now need to handle those servlet mechanism.

This last one is a little strange, as we normally avoid calling a Callback passed to a handler until the request/response processing is truly complete. But since we don't have a callback passed to the Servlet, we have to infer if it is complete or not by looking at the state. If it is still in ERRORING state, then the servlet did not change state and we are complete, else we are starting a new servlet dispatch lifecycle!   Hmm I wonder if the ErrorHandler that dispatches to  should really do this. Hmmm but it doesn't have ServletChannelState access?

Also need to document that the `failure` is a failure generated when calling the errorHandler.   Perhaps name it `errorHandlerFailure`?

Reply:
Starting a new Servlet lifecycle from an `ErrorHandler` is fine and we have tests, see ee10 `ErrorPageTest`.